crates/router/src/db/merchant_account.rs

```diff
@@ -513,11 +513,29 @@ async fn publish_and_redact_merchant_account_cache(
         .as_ref()
         .map(|publishable_key| CacheKind::Accounts(publishable_key.into()));
 
+    #[cfg(feature = "business_profile_routing")]
+    let kgraph_key = merchant_account.default_profile.as_ref().map(|profile_id| {
+        CacheKind::CGraph(
+            format!(
+                "kgraph_{}_{}",
+                merchant_account.merchant_id.clone(),
+                profile_id,
+            )
+            .into(),
+        )
+    });
+
+    #[cfg(not(feature = "business_profile_routing"))]
+    let kgraph_key = Some(CacheKind::CGraph(
+        format!("kgraph_{}", merchant_account.merchant_id.clone()).into(),
+    ));
```

Comment:
Same here, should the variable name have been `cgraph_key` and the format string `cgraph_{}`?

Why don't we consider adding a function to generate the cgraph cache key for a specified merchant ID, and an optionally specified business profile ID (the profile ID could also be specified based on the `business_profile_routing` being enabled)?